### PR TITLE
arrays

### PR DESCRIPTION
**Pull request by DzmitryKosmach**

### CODE REVIEW

File: task_arrays/index.html [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<html>
+  <head>
+    <title>Arrays</title>
+    <script src="js/main.js" type="text/javascript"></script>
+  </head>
+  <body>
+    Arrays
+  </body>
+</html>
```
Comment (NikitaSedyx) [inline task_arrays/index.html]: nit: empty line at the eof

File: task_arrays/js/main.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+function last(array) {
+  return array[array.length - 1];
+}
+
+function excludeLast(array, exclude = 1) {
+  if (exclude > 0) {
+    array.length -= exclude;
+  }
+}
```
Comment (NikitaSedyx) [inline task_arrays/js/main.js]: same

File: task_arrays/js/main.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+  return uniqueArray;
+}
+
+function last(array) {
+  return array[array.length - 1];
+}
+
+function excludeLast(array, exclude = 1) {
+  if (exclude > 0) {
+    array.length -= exclude;
```
Comment (NikitaSedyx) [inline task_arrays/js/main.js]: you should remember, it mutates array, -1

File: task_arrays/js/main.js [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+function sumReduce(array) {
+  return array.reduce(function (sum, current) {
+    return sum + current;
+  });
+}
+
+function unique(array) {
+  var uniqueArray = [];
+  array.forEach(function () {
```
Comment (NikitaSedyx) [inline task_arrays/js/main.js]: what is the problem to add function arguments instead of usage `arguments`

File: task_arrays/js/main.js [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+  for (var i = startValue; i < endValue; i += stepValue) {
+    array.push(i);
+  }
+  return array;
+}
+
+function compactFor(array) {
+  var compactArray = [];
+  for (var i = 0; i < array.length; i++) {
+    if (!!array[i]) {
```
Comment (NikitaSedyx) [inline task_arrays/js/main.js]: what do you think will `if (array[i])` work without `!!`? ad why

File: task_arrays/js/main.js [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+}
+
+function isObject(arg) {
+  if (arg === null) {
+    return false;
+  }
+  return typeof arg === 'object';
+}
+
+function range() {
```
Comment (NikitaSedyx) [inline task_arrays/js/main.js]: next cases should work:
range(10) [0..9]
range(1, 10) [1..9]
range(1, 10, 3) [1, 4, 7]
range(10, null, 3) [0, 3, 6, 9]
range(10, null) the same with range(10)
range(-2, -5) []
range(-5, -2) [-5, -4, -3]
range(-5, -2, 2) [-5, -3]
range(-5, -2, -2) []
range(-5, null, 2) []
range(-5, null, -2) [0, -2, -4]
range(-10, -20, -5) [-10, -15]
range(-20, -10, -5) []

FYI this function can be implemented ~10 lines, firstly you should find step and length of result array (the difficult part of this function)

Comment (NikitaSedyx) [inline task_arrays/js/main.js]: where are you function arguments?

**Review by NikitaSedyx — CHANGES_REQUESTED** [review]

**Comment by DzmitryKosmach** [comment]
https://github.com/DzmitryKosmach/DzmitryKosmach.github.io/pull/4/commits/3d570767986f7ea9c410466d28864d4328ed2c6f


2017-04-12 7:44 GMT+03:00 Mikita Siadykh <notifications@github.com>:

File: task_arrays/js/main.js [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+function sumReduce(array) {
+  return array.reduce(function (sum, current) {
+    return sum + current;
+  });
+}
+
+function unique(array) {
+  var uniqueArray = [];
+  array.forEach(function (value, index) {
+    if (uniqueArray.every(function (item) {
```
Comment (NikitaSedyx) [inline task_arrays/js/main.js]: I'd suggest to move `every` outside of `if` to better readability

**Review by NikitaSedyx — APPROVED** [review]
mostly lgtm

**Pull request merged**